Libplanet/Net/Swarm.cs

```diff
@@ -1339,11 +1339,21 @@ CancellationToken cancellationToken
             CancellationToken cancellationToken)
         {
             HashDigest<SHA256> genesisHash = BlockChain.Genesis.Hash;
-            IComparer<IBlockExcerpt> canonComparer = BlockChain.Policy.CanonicalChainComparer;
+            IComparer<BlockPerception> canonComparer = BlockChain.Policy.CanonicalChainComparer;
+            BlockPerception tipPerception = BlockChain.PerceiveBlock(initialTip);
             var peersWithHeightAndDiff = (await DialToExistingPeers(dialTimeout, cancellationToken))
                 .Where(pp =>
-                    genesisHash.Equals(pp.ChainStatus?.GenesisHash) &&
-                    canonComparer.Compare(pp.ChainStatus, initialTip) > 0)
+                {
+                    if (!(pp.ChainStatus is ChainStatus chainStatus))
+                    {
+                        return false;
+                    }
+
+                    return genesisHash.Equals(chainStatus.GenesisHash) && canonComparer.Compare(
+                        BlockChain.PerceiveBlock(chainStatus, chainStatus.Timestamp),
+                        tipPerception
+                    ) > 0;
```

Comment:
Don't we need to check `Block<T>.ProtocolVersion` for backward compatibility at this moment?

Reply:
This does not change the consensus for (kinda) “finalized” blocks.  With or without version check, old enough blocks will be considered as they had been.